cdap-api/src/main/java/co/cask/cdap/internal/api/AbstractProgramDatasetConfigurable.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.api;
+
+import co.cask.cdap.api.DatasetConfigurer;
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+
+/**
+ * Helper methods to help add stream/dataset to Programs.
+ *
+ * @param <T> Program's configurer
+ */
+public abstract class AbstractProgramDatasetConfigurable<T extends DatasetConfigurer> {
```

Comment:
@chtyim But Abstract{Flow, MapReduce, etc} extend this class. So will that violate the compatibility guarantees?